### ISSUE TITLE
428: DeprecationWarning: use loseWriteConnection instead

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3220]
|<img alt="JimD's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| JimD reported|
|-|-|
|Trac ID|trac#3220|
|Type|defect|
|Created|2008-05-02 08:12:47Z|
|Branch|https://github.com/twisted/twisted/tree/conch-losewriteconnection-3220|

Regarding line 428 of: 

http://twistedmatrix.com/trac/browser/trunk/twisted/conch/scripts/conch.py?rev=23479

Apparently fixed by:

```
--- /usr/local/src/conch.py     2008-05-02 17:02:11.000000000 -0700
+++ /usr/lib/python2.5/site-packages/twisted/conch/scripts/conch.py     2006-01-17 12:49:40.000000000 -0800
@@ -425,7 +425,7 @@
 
     def eofReceived(self):
         log.msg('got eof')
-        self.stdio.loseWriteConnection()
+        self.stdio.closeStdin()
     
     def closeReceived(self):
         log.msg('remote side closed %s' % self)
```

<details><summary>Searchable metadata</summary>

```
trac-id__3220 3220
type__defect defect
reporter__JimD JimD
priority__normal normal
milestone__ 
branch__branches_conch_losewriteconnection_3220 branches/conch-losewriteconnection-3220
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__DeprecationWarning DeprecationWarning
time__1209715967000000 1209715967000000
changetime__1231436555000000 1231436555000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="launchpad's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|launchpad commented|
|-|-|

Updating Launchpad bug reference

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

fixing description markup

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

The patch is backward.  It's `closeStdin` that's used and deprecated and `loseWriteConnection` which ought to be used instead.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [25757]) Branching to 'conch-losewriteconnection-3220'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun removed owner|
|-|-|

Build results:

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/conch-losewriteconnection-3220

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @exarkun|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [25881]) Merge conch-losewriteconnection-3220

Author: JimD, exarkun
Reviewer: therve
Fixes: [#3220](https://github.com/twisted/twisted/issues/3220)

Replace the use of the deprecated StandardIO.closeStdin with use of the
equivalent StandardIO.loseWriteConnection.  This eliminates a deprecation
warning when a Conch SSH session ends.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
